### ISSUE TITLE
[UI] Cluster Teamplate Issues - Override switch missing for Private Registry

### ISSUE THREAD

**Issue by davidnuzik** [rancher/rancher#21567]
- [ ] **_Override switch missing for Private Registry when it's disabled:_**

1. Similar to above issue, Private Registry radio buttons are Disabled by default. It's confusing because I still expect to see an override button but cannot until I enable it.

RESULT: This is also confusing to the user. Private Registry is Disabled by default but I should still see the override switch.
![image](https://user-images.githubusercontent.com/45179589/60928606-f0165180-a262-11e9-8136-406576ce3884.png)

**Comment by davidnuzik** [comment]
Version: master-head (8/7/19)

The override switch has been added in the UI when creating a CT:
![image](https://user-images.githubusercontent.com/45179589/62651006-e5f07d00-b90c-11e9-9ee9-b1d9f2f238db.png)

However, when you attempt to utilize this it does not render correctly in the UI. Pictured below is me trying to leverage my CT which sets Private Registry to Enabled (with override) and with override for each URL, User, and Password, but with nothing filled in for those fields.

For some reason we have two entries for Private registry (should not be possible only one is allowed) and I can't change from "Enabled" to "Disabled" (I should be able to since I allowed override for "Private Registry"
![image](https://user-images.githubusercontent.com/45179589/62651220-54cdd600-b90d-11e9-8a9e-3f3c6dd1f1ac.png)

**Comment by davidnuzik** [comment]
Version: master-head (8/8/19)

I can confirm that we render the Private Registry section in the UI correctly now when using a cluster template that has an override (and enabled) for Private registry. See https://github.com/rancher/rancher/issues/21567#issuecomment-519233476 comment for details about that problem.

![image](https://user-images.githubusercontent.com/45179589/62726700-b143fa80-b9cc-11e9-93e7-6de243fe58e9.png)

I performed other standard tests such as testing both with and without overrides set. Setting overrides for some but not all fields in this section, and etc. This is working as expected now as far as proper rendering in the UI when leveraging the template.

I also made sure the CT works as expected when filling in data in the CT and leveraging the CT (with and without overrides).

I found just one minor issue such that url and username are not displayed when leveraging the template and created a separate issue to track this here: https://github.com/rancher/rancher/issues/22072
